### ISSUE TITLE
SSH tunnel forwarding needs to handle alternate proxy URLs

### ISSUE THREAD

**Issue by brendandburns** [kubernetes/kubernetes#17172]
There are two valid proxy URLs:

http://localhost:8001/api/v1/proxy/namespaces/default/pods/nginx-y0iu3:80/

and

http://localhost:8001/api/v1/namespaces/default/pods/nginx-y0iu3:80/proxy/

When we are using SSH tunnels for proxying, the latter URL doesn't work.

We need to add an e2e test for this and fix the code.

@cjcullen @lavalamp

**Comment by liggitt** [comment]
Was this seen in master or 1.1?

**Comment by liggitt** [comment]
There is an e2e test in master for this, which should be exercised in gke
Fixed in master in https://github.com/kubernetes/kubernetes/pull/15224
Cherry pick to 1.1 is https://github.com/kubernetes/kubernetes/pull/17127

**Comment by michaelukpong** [comment]
@liggit this was seen on GKE...

**Comment by liggitt** [comment]
On master or on 1.1?

**Comment by michaelukpong** [comment]
@liggitt  not sure how to check that.... my understanding --> if I install kubernetes locally then ofcourse i can chose which release to git pull/download.

But on Google cloud... i just simply create a cluster and keep doing stuff.... how can i check which kubernetes Google cloud is running on GKE? ...

Thanks and forgive my novice knowledge on this subject matter.

**Comment by liggitt** [comment]
https://$MASTER/version should report the build

**Comment by michaelukpong** [comment]
okay thanks 
```
{
major: "1",
minor: "0",
gitVersion: "v1.0.7",
gitCommit: "6234d6a0abd3323cd08c52602e4a91e47fc9491c",
gitTreeState: "clean"
}
```

**Comment by jlowdermilk** [comment]
@michaelukpong unfortunately the cherry pick of the fix missed the 1.1 release cut. It should be included in the 1.1.2 patch release.

As an aside, you can get server/client version info for kubernetes on any platform by running
```
kubectl version
```

**Comment by a-robinson** [comment]
Looks like this should be closed :)
